docs/G-Codes.md

```diff
@@ -642,14 +642,50 @@ section is enabled:
 
 ## Adxl345 Accelerometer Commands
 
-The following command is available when an "adxl345" config section is
+The following commands are available when an "adxl345" config section is
 enabled:
 - `ACCELEROMETER_MEASURE [CHIP=<config_name>] [RATE=<value>]
   [NAME=<value>]`: Starts accelerometer measurements at the requested
   number of samples per second. If CHIP is not specified it defaults
   to "default". Valid rates are 25, 50, 100, 200, 400, 800, 1600,
-  and 3200. If RATE is zero (or not specified) then the current series
-  of measurements are stopped and the results are written to a file
-  named `/tmp/adxl345-<name>.csv` where "<name>" is the optional NAME
-  parameter. If NAME is not specified it defaults to the current time
-  in "YYYYMMDD_HHMMSS" format.
+  and 3200. The command works in a start-stop mode: when executed for
+  the first time, it starts the measurements, next execution stops them.
+  If RATE is not specified, then the default value is used (either from
+  `printer.cfg` or `3200` default value). The results of measurements
+  are written to a file named `/tmp/adxl345-<name>.csv` where "<name>"
+  is the optional NAME parameter. If NAME is not specified it defaults
+  to the current time in "YYYYMMDD_HHMMSS" format.
+- `ACCELEROMETER_QUERY [CHIP=<config_name>] [RATE=<value>]`: queries
+  accelerometer for the current value. If CHIP is not specified it
+  defaults to "default". If RATE is not specified, the default value is
+  used. This command is useful to test the connection to the ADXL345
+  accelerometer: one of the returned values should be a free-fall
+  acceleration (+/- some noise of the chip).
+
+## Resonance Testing Commands
+
+The following commands are available when a "resonance_tester" config section
+is enabled:
+- `MEASURE_AXES_NOISE`: Measures and outputs the noise for all axes of all
+  enabled accelerometer chips.
+- `TEST_RESONANCES AXIS=<axis> [CSV_NAME=<csv_file>] [RAW_NAME=<raw_cvs_file>]
+  [FREQ_START=<min_freq>] [FREQ_END=<max_freq>] [HZ_PER_SEC=<hz_per_sec>]`:
+  Runs the resonance test in all configured probe points for the requested
+  axis (X or Y) and measures the acceleration using the accelerometer chips
+  configured for the respective axis. If `RAW_NAME` is set, then the raw
+  accelerometer data is written into a file or a series of files
+  `/tmp/<raw_csv_name>_<axis>_<point>_YYYYMMDD_HHMMSS<raw_csv_ext>`.
+  If `CSV_NAME` is set, it is treated as a template
+  `<csv_file> = <csv_name>+<csv_ext>`; the frequency response is calculated
+  and written into `/tmp/<csv_name>_<axis>_YYYYMMDD_HHMMSS<csv_ext>` file.
+  At least one of the `CSV_NAME` or `RAW_NAME` parameters must be provided.
```

Comment:
I was thinking that it might be too easy to overwrite an existing file if the user forgets to update the file name (in repeated measurements). So, for example, `AXIS=Y RAW_NAME=resonances.csv` will produce a file `/tmp/resonances_x_YYYYMMDD_HHMMSS.csv` or, if the multiple test points were configured, e.g. to `/tmp/resonances_x_100.000_100.000_20.000_YYYYMMDD_HHMMSS.csv`. At least, some template expansion is likely necessary (as opposed to just take `RAW_NAME` value) in case multiple test points were configured (an alternative would be to concatenate the measurements into a single file, but that's not great, in my opinion). So, Kevin, what would you prefer: make this command resemble `ACCELEROMETER_MEASURE` (but then what to do with raw data from multiple test points?), maybe try to clarify the description and/or add a some examples how the file name will be expanded?

Separately, I am going to add a confirmation message following @Sineos suggestion: just print the file name of the file written. So maybe that will also help to avoid the confusion.

Reply:
I guess I'd say use a hard-coded template expansion, but allow the user only specify a "name" that (if not specified) defaults to YYYYMMDD_HHMMSS.  So, for example, if the user specified `NAME=foo` then an output file might look like: `/tmp/resonances_x_100.000_100.000_20.000_foo.csv`.

Reply:
I made the changes along those lines.